### ISSUE TITLE
[CI] Leaked Files Failure in org.elasticsearch.index.engine.FrozenEngineTests.testSearchers

### ISSUE THREAD

**Issue by original-brownbear** [elastic/elasticsearch#68642]
This just failed in https://gradle-enterprise.elastic.co/s/l6g7t7wff2nvk/console-log/raw on master.

```
./gradlew ':x-pack:plugin:frozen-indices:test' --tests "org.elasticsearch.index.engine.FrozenEngineTests.testSearchers" -Dtests.seed=B276FDBDB4EB162 -Dtests.security.manager=true -Dtests.locale=bg -Dtests.timezone=NET -Druntime.java=11
```

Fails as:

```
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:frozen-indices:test' --tests "org.elasticsearch.index.engine.FrozenEngineTests.testSearchers" -Dtests.seed=B276FDBDB4EB162 -Dtests.security.manager=true -Dtests.locale=bg -Dtests.timezone=NET -Druntime.java=11
  2> java.lang.RuntimeException: MockDirectoryWrapper: cannot close: there are still 33 open files: {_6_1_Asserting_0.dvd=1, _7.cfs=1, _c_1_Asserting_0.dvd=1, _10.cfs=1, _1a.cfs=1, _19.cfs=1, _1_1_Asserting_0.dvd=1, _6.cfs=1, _11.cfs=1, _5.cfs=1, _2_1_Asserting_0.dvd=1, _12.cfs=1, _e.cfs=1, _8_1_Asserting_0.dvd=1, _13.cfs=1, _3.cfs=1, _f.cfs=1, _c.cfs=1, _3_1_Asserting_0.dvd=1, _14.cfs=1, _2.cfs=1, _7_3_Asserting_0.dvd=1, _15.cfs=1, _0_2_Asserting_0.dvd=1, _1.cfs=1, _d.cfs=1, _0.cfs=1, _5_1_Asserting_0.dvd=1, _9.cfs=1, _18.cfs=1, _b.cfs=1, _1b.cfs=1, _8.cfs=1}
        at __randomizedtesting.SeedInfo.seed([B276FDBDB4EB162:81A9CD1F24D9274B]:0)
        at org.apache.lucene.store.MockDirectoryWrapper.close(MockDirectoryWrapper.java:843)
        at org.apache.lucene.store.FilterDirectory.close(FilterDirectory.java:110)
        at org.apache.lucene.store.FilterDirectory.close(FilterDirectory.java:110)
        at org.elasticsearch.index.store.Store$StoreDirectory.innerClose(Store.java:711)
        at org.elasticsearch.index.store.Store.closeInternal(Store.java:415)
        at org.elasticsearch.index.store.Store$1.closeInternal(Store.java:143)
        at org.elasticsearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:52)
        at org.elasticsearch.index.store.Store.decRef(Store.java:390)
        at org.elasticsearch.index.store.Store.close(Store.java:397)
        at org.elasticsearch.index.engine.FrozenEngineTests.testSearchers(FrozenEngineTests.java:377)

        Caused by:
        java.lang.RuntimeException: unclosed IndexInput: _3.cfs
```

Looking into it.

**Comment by original-brownbear** [comment]
Reproduces locally with master after a few hundred runs with the seed in the issue.

**Comment by original-brownbear** [comment]
This is caused by a Lucene bug https://github.com/apache/lucene-solr/commit/47e3d06ce00642624634e5d45ebc16fa33d48099 that was recently fixed but apparently the fix didn't go into 8.8.0 that we now use in `master`.

**Comment by original-brownbear** [comment]
Nhat informed me that this will be fixed shortly in Lucene 8.8.1 so I'm closing here as it's a rare failure.
